pkg/server/api/agent/v1/service.go

```diff
@@ -658,17 +674,23 @@ func validateAttestAgentParams(params *agentv1.AttestAgentRequest_Params) error
 	}
 }
 
-func getAttestAgentResponse(spiffeID spiffeid.ID, certificates []*x509.Certificate) *agentv1.AttestAgentResponse {
+func getAttestAgentResponse(spiffeID spiffeid.ID, certificates []*x509.Certificate, canReattest bool) *agentv1.AttestAgentResponse {
 	svid := &types.X509SVID{
 		Id:        api.ProtoFromID(spiffeID),
 		CertChain: x509util.RawCertsFromCertificates(certificates),
 		ExpiresAt: certificates[0].NotAfter.Unix(),
 	}
 
+	var reattestable bool
+	if fflag.IsSet(fflag.FlagReattestToRenew) {
```

Comment:
Thinking about it more... I think this is the wrong spot for this flag test. We should probably persist the response given by the server but only act on it if the flag is set. In other words, the rotation code that either renew's or re-attests should do this test IMO.

Reply:
I'll make that change. Downside with that approach is that you have to set the feature flag on agent and server. With this approach you only have to set it on the server side.

Reply:
I think you'll need to have the flag in both places in order to deploy this into an existing deployment safely. Since the server side flag will reject agents who try and renew without reattestation, you can't turn on the server side flag until all agents have been deployed AND have gone through a renewal flow in order to record the correct re-attestable value so they can do the expected operation at renew/re-attest time.

Reply:
makes sense, thats a lot better flow to upgrade. this change is in.